tests/components/vesync/common.py

```diff
@@ -1,7 +1,30 @@
 """Common methods used across tests for VeSync."""
 import json
 
-from tests.common import load_fixture
+import requests_mock
+
+from homeassistant.components.vesync.const import DOMAIN
+
+from tests.common import load_fixture, load_json_object_fixture
+
+ALL_DEVICES = load_json_object_fixture("vesync-devices.json", DOMAIN)
+ALL_DEVICE_NAMES: list[str] = [
+    dev["deviceName"] for dev in ALL_DEVICES["result"]["list"]
+]
+
+
+def mock_devices_response(
+    requests_mock: requests_mock.Mocker, device_name: str
+) -> None:
+    """Build a response for the Helpers.call_api method."""
+    device_list = []
+    for device in ALL_DEVICES["result"]["list"]:
+        if device["deviceName"] == device_name:
+            device_list.append(device)
+    requests_mock.post(
+        "https://smartapi.vesync.com/cloud/v1/deviceManaged/devices",
+        json={"code": 0, "result": {"list": device_list}},
+    )
```

Comment:
And here you can inject the fixture corresponding to each `device_name`.
```suggestion
    )
    requests_mock.post(
        "https://smartapi.vesync.com/cloud/v1/user/login",
        json=load_json_object_fixture("vesync-login.json", DOMAIN),
    )
    for fixture in DEVICE_FIXTURES[device_name]:
        requests_mock.request(
            fixture[0],
            f"https://smartapi.vesync.com{fixture[1]}",
            json=load_json_object_fixture(fixture[2], DOMAIN),
        )
```